gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcDirectServerStreamingCallable.java

```diff
@@ -54,20 +60,120 @@
   }
 
   @Override
-  public void serverStreamingCall(
-      RequestT request, ApiStreamObserver<ResponseT> responseObserver, ApiCallContext context) {
+  public void call(
+      RequestT request, ResponseObserver<ResponseT> responseObserver, ApiCallContext context) {
     Preconditions.checkNotNull(request);
     Preconditions.checkNotNull(responseObserver);
 
     ClientCall<RequestT, ResponseT> call = GrpcClientCalls.newCall(descriptor, context);
-    ClientCalls.asyncServerStreamingCall(
-        call, request, new ApiStreamObserverDelegate<>(responseObserver));
+    GrpcStreamController<RequestT, ResponseT> controller =
+        new GrpcStreamController<>(call, responseObserver);
+    controller.start(request);
   }
 
-  @Override
-  public Iterator<ResponseT> blockingServerStreamingCall(RequestT request, ApiCallContext context) {
-    Preconditions.checkNotNull(request);
-    ClientCall<RequestT, ResponseT> call = GrpcClientCalls.newCall(descriptor, context);
-    return ClientCalls.blockingServerStreamingCall(call, request);
+  /**
+   * Wraps a GRPC ClientCall in a {@link StreamController}. It feeds events to a {@link
+   * ResponseObserver} and allows for back pressure.
+   */
+  static class GrpcStreamController<RequestT, ResponseT> extends StreamController {
```

Comment:
I have a preference for top-level types over embedded types, could this and the next class be moved to their own files?